src/main/scala-2.12/org/apache/james/gatling/jmap/scenari/JmapMessageFlagUpdatesScenario.scala

```diff
@@ -0,0 +1,26 @@
+package org.apache.james.gatling.jmap.scenari
+
+import io.gatling.core.Predef._
+import io.gatling.core.structure.ScenarioBuilder
+import org.apache.james.gatling.control.UserFeeder.UserFeederBuilder
+import org.apache.james.gatling.jmap.{CommonSteps, JmapMessages}
+
+class JmapMessageFlagUpdatesScenario {
+  private object Keys {
+    val inbox = "inboxID"
+    val messageIds = "messageIds"
+    val messagesDetailList = "messagesDetailList"
+  }
+
+  def generate(userFeeder: UserFeederBuilder): ScenarioBuilder = {
+    scenario("JmapMessageFlagUpdatesScenario")
+      .feed(userFeeder)
+      .exec(CommonSteps.authentication())
+      .group(MessageFlagUpdates.name)(
+        randomSwitch(
+          70.0 -> exec(JmapMessages.markAsRead()),
```

Comment:
To be calling such an action, one need to have a `messageIds` variable persisted in the session.

This is done by doing a getMessageList before.

You need to add these steps before your flag update random switch:

```
.exec(RetryAuthentication.execWithRetryAuthentication(JmapMailbox.getMailboxes, JmapMailbox.getMailboxesChecks ++ JmapMailbox.saveInboxAs(Keys.inbox)))       

.exec(RetryAuthentication.execWithRetryAuthentication(JmapMessages.listMessages(openpaasListMessageParameters(Keys.inbox)), JmapMessages.listMessagesChecks)))
```